src/tools/perf/perftest_params.c

```diff
@@ -756,9 +783,14 @@ ucs_status_t adjust_test_params(perftest_params_t *params,
     }
 
     if (params->super.send_device.mem_type != UCS_MEMORY_TYPE_LAST) {
+        const char *channel_str = getenv("UCX_RC_GDA_NUM_CHANNELS");
+        if (channel_str) {
+            params->super.device_num_channels = atoi(channel_str);
+        }
```

Comment:
seems very weird, why is this needed?
we can pass arbitrarily large channel id to the UCP/UCT device APIs, and the transport should anyway do % operation to select the right channel according to the number of QPs

Reply:
UCP/UCT doesn't do %. Perftest needs to know how many channels created and do % at user-app side.
Maybe add an option for perftest and adjust ucp config in perftest? Or any better suggestions?